src/services/user.rs

```diff
@@ -178,3 +187,121 @@ pub fn auth_and_delete_integration<T: user::UserModel + integration::Integration
     let _ = jwt::verify_token(Utc::now().naive_utc(), token, jwt_secret)?;
     Ok(database.delete_integration(&id)?)
 }
+
+#[cfg(test)]
+mod user_tests {
+
+    use super::*;
+
+    use crate::entities::user;
+
+    struct T {}
+
+    impl user::UserModel for T {
+        fn create_user(&self, user: user::NewUser) -> user::Result<user::User> {
+            Ok(user::User {
+                id: Uuid::from_u128(160141200314647599499076565412518613020),
+                name: user.name,
+                username: user.username,
+                register_date: None,
+                email: user.email,
+                last_code_gen_request: None,
+                login_code: None,
+                is_registered: true,
+                payday: None,
+            })
+        }
+        fn delete_user(&self, id: &Uuid) -> user::Result<user::User> {
+            Ok(user::User {
+                id: Uuid::from_u128(160141200314647599499076565412518613020),
+                name: "Usuário 1".to_string(),
+                username: "usuario1".to_string(),
+                register_date: None,
+                email: "usuario1@gmail.com".to_string(),
+                last_code_gen_request: None,
+                login_code: None,
+                is_registered: true,
+                payday: None,
+            })
+        }
+        fn get_user(&self, id: Uuid) -> user::Result<user::User> {
+            Ok(user::User {
+                id: Uuid::from_u128(160141200314647599499076565412518613020),
+                name: "Usuário 1".to_string(),
+                username: "usuario1".to_string(),
+                register_date: None,
+                email: "usuario1@gmail.com".to_string(),
+                last_code_gen_request: None,
+                login_code: None,
+                is_registered: true,
+                payday: None,
+            })
+        }
+        fn check_if_username_available(&self, username: &str) -> user::Result<bool> {
+            match username {
+                "usuario1" => Ok(true),
+                "usuario2" => Ok(true),
+                "usuario3" => Ok(false),
+                "usuario4" => Ok(false),
+                _ => Ok(true),
```

Comment:
In match expressions, you can match multiple patterns using the | syntax.
```suggestion
            match username {
                "usuario1" | "usuario2" | _ => Ok(true),
                "usuario4" | "usuario3" => Ok(false),
```

Reply:
Done